plugins/modules/zos_copy.py

```diff
@@ -2040,16 +2084,22 @@ def allocate_destination_data_set(
             size = int(src_attributes.total_space)
             record_format = src_attributes.recfm
             record_length = int(src_attributes.lrecl)
-
-            dest_params = get_data_set_attributes(dest, size, is_binary, record_format=record_format, record_length=record_length, type="PDSE", volume=volume)
+            dest_params = get_data_set_attributes(dest, size, is_binary, record_format=record_format, record_length=record_length, type="PDSE",
+                                                  volume=volume)
             data_set.DataSet.ensure_present(replace=force, **dest_params)
         elif src_ds_type == "USS":
```

Comment:
note that my version of the updated doc for the `dest` option explains this behavior:
```
      - When C(dest) is a data set, precedence rules apply. If C(dest_data_set)
        is set, this will take precedence over an existing data set. If C(dest)
        is an empty data set, the empty data set will be written with the
        expectation its attributes satisfy the copy. If C(dest) does not exist and 
        I(is_executable) has been set, C(dest) will have an Undefined (U) record format,
        with a record length of 0, block size of 32760 and the remaining attributes will be 
        computed. Lastly, if no precedence rule has been exercised, C(dest) will be created
        with the same attributes of C(src).
```

Basically, I've added a precedence rule above creating `dest` based on `src` attrs